README.md

```diff
@@ -63,6 +66,45 @@ $ sudo docker run --rm -ti -v "$(realpath .):/app" --user $(id -u $USER):$(id -g
 root@656be163fe84:/app# BOLOS_SDK=$NANOS_SDK make scan-build
 ```
 
+## App testing
+
+With the `ledger-app-dev-tools` image, whether you are developing on macOS, Windows or Linux, you can quickly test your app with the [Speculos](https://github.com/LedgerHQ/speculos) emulator or the [Ragger](https://github.com/LedgerHQ/ragger) test framework.
+For examples of functional tests implemented with Ragger, you can have a look at the [app-boilerplate](https://github.com/LedgerHQ/app-boilerplate)
+
+First, run the `ledger-app-dev-tools` docker image. Depending on your platform, the command will change slightly :
+
+**Linux (Ubuntu)**
+
+```bash
+sudo docker run --rm -ti -v "$(realpath .):/app" --user $(id -u):$(id -g) -v "/tmp/.X11-unix:/tmp/.X11-unix" -e DISPLAY=$DISPLAY ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools:latest
+```
+
+**Windows (with PowerShell)**
+
+Assuming you already have a running X server like [VcXsrv](https://sourceforge.net/projects/vcxsrv/).
+
+```bash
+$fullPath = Resolve-Path .
+docker run --rm -ti -v "${fullPath}:/app" -e DISPLAY="host.docker.internal:0" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools:latest
+```
+
+**macOS**
+
+Assuming you already have a running X server like [XQuartz](https://www.xquartz.org/) configured to accept client connections.
+
+```bash
+sudo docker run --rm -ti -v "$(pwd -P):/app" --user $(id -u):$(id -g) -v "/tmp/.X11-unix:/tmp/.X11-unix" -e DISPLAY="host.docker.internal:0" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools:latest
+```
+
+Then you can test your app :
+
+```bash
+# Run your app on Speculos
+root@656be163fe84:/app# speculos build/nanos/bin/app.elf --model nanos
+# Run ragger functional tests (assuming you have some implemented)
+root@656be163fe84:/app# pytest tests/ --tb=short -v --device nanos --display
```

Comment:
Is there a way to send apdu to such a speculos instance?
Maybe somethink like running this in background and using command like this:
`echo ed060000080000000461626364 | LEDGER_PROXY_ADDRESS=127.0.0.1 LEDGER_PROXY_PORT=9999 python3 -m ledgerblue.runScript --apdu`
Maybe we could document it here?

also having the two commands (speculoas and ragger) in the same snippet feels like speculos need to be running for ragger to be launch, which is not the case.

Reply:
Didn't try what you suggest for APDUs. Good thing to try but do you want us to mention it here ? Maybe more suited for app-boilerplate things.

As for the commands, I'll try to make it clearer they are not related.